### PR TITLE
Add tests and match conventions

### LINKED ISSUE #2: Add tests and match conventions
1. Add integration tests to check the commands.
2. Add workflows for running the tests on releases.
3. Update the names of the `makefile` targets to match convention.
4. Add `fybrik.io` prefix to the module name.

### PR DESCRIPTION
**Pull request by Vicky-Goldin**
1. Added integration tests that check the `validate` and `compile` commands.
2. Updated the workflows to include running the tests and `golint`.
3. Updated the names of the makefile targets to match convention.
4. Added `fybrik.io` prefix to the module name.
5. Removed the `pkg/validate/resource.go` file that wasn't used in this repository.

Closes #2

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -35,9 +50,9 @@ jobs:
         echo ::set-output name=version::$version
         echo "DOCKER_TAG=${version}" >> $GITHUB_ENV
     - name: Build docker image
```
Comment (roytman) [inline .github/workflows/push.yml:52]: the comment at Line 36 is not relevant, we don't create helm charts here

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -35,9 +50,9 @@ jobs:
         echo ::set-output name=version::$version
```
Comment (roytman) [inline .github/workflows/push.yml:50]: The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Review by roytman — COMMENTED** [review]
Thanks for the PR

File: pkg/validate/structural.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ package validate
 import (
 	"context"
 
-	"taxonomy-cli/pkg/taxonomyio"
+	"fybrik.io/taxonomy-cli/pkg/taxonomyio"
```
[the remark this answers is not in the document]
Comment (Vicky-Goldin) [reply pkg/validate/structural.go:9]: Fixed

File: Makefile [hunk 4]
```diff
@@ -1,25 +1,32 @@
...
 
-.PHONY: push-docker
-push-docker:
+.PHONY: docker-push
+docker-push:
 	docker push $(DOCKER_IMG)
+
+.PHONY: test
+test: local-build
+	go test -v ./...
```
[the remark this answers is not in the document]
Comment (Vicky-Goldin) [reply Makefile:31]: Done

File: .github/workflows/golangci-lint.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+          fetch-depth: 0
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+          args: --new-from-rev=HEAD~
```
[the remark this answers is not in the document]
Comment (Vicky-Goldin) [reply .github/workflows/golangci-lint.yml:24]: Done, with some code modification to pass the linter.

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -10,11 +10,26 @@ on:
 env:
   DOCKER_HOSTNAME: ghcr.io
   DOCKER_USERNAME: ${{ github.actor }}
+  GO_VERSION: 1.19
 
 jobs:
+  test:
```
[the remark this answers is not in the document]
Comment (Vicky-Goldin) [reply .github/workflows/push.yml:16]: Done

File: .github/workflows/push.yml (hunk 1 above)
Comment (Vicky-Goldin) [reply .github/workflows/push.yml:52]: The `DOCKER_TAG` env is used later in the makefile as the tag of the docker image that is build and published, so I think that without this line the tag will always be `main` as the default value from the makefile.

Comment (roytman) [reply .github/workflows/push.yml:52]: I meant the comment about helm

File: Makefile (hunk 4 above)
[the remark this answers is not in the document]
Comment (roytman) [reply Makefile]: it should be `go test -v ./... -ginkgo.v -ginkgo.progress -test.v`

File: pkg/validate/structural.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (roytman) [reply pkg/validate/structural.go]: this repo should be the last

**Review by roytman — APPROVED** [review]

**Pull request merged**